### PR TITLE
Add --custom-function parameter to check

### LINKED ISSUE #5: Add --custom-function parameter to check
In one of the projects that I have been working for a while, there are some legacy custom dump functions created and used frequently by the team. A sample function is;

`function pre_dump( $var ) { 
    echo '<pre>'; 
    var_dump( $var ); 
    echo '</pre>'; 
}
`
The check tool can not find the usage of such kind of functions. A sample usage might be;

`$ ./vendor/bin/var-dump-check --no-colors --tracy --custom-function pre_dump,echo_dump .`

### PR DESCRIPTION
**Pull request by umutphp**
This pull request is the clone of [the pull request](https://github.com/JakubOnderka/PHP-Var-Dump-Check/pull/32) for the previous repository ([JakubOnderka/PHP-Var-Dump-Check](https://github.com/JakubOnderka/PHP-Var-Dump-Check)). 

Closes #5

### CODE REVIEW

File: src/Settings.php [hunk 1]
```diff
@@ -138,7 +138,7 @@ public static function parseArguments(array $arguments)
         $setting->functionsToCheck = array_unique($setting->functionsToCheck);
 
         // Any custom function is given
-        if (count($customFunctions) > 0) {
+        if (!empty($customFunctions)) {
```
Comment (grogy) [inline src/Settings.php]: Please, why here is used `empty`?

File: README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ Options for run
 - `--doctrine` - check dump: `Doctrine::dump`, `\Doctrine\Common\Util\Debug::dump`
 - `--symfony` - check dump: `dump`, `VarDumper::dump`, `VarDumper::setHandler`
 - `--laravel` - check dump: `dd`, `dump`
-- `--custom-function` - comma separated custom function name(s) to check like "pre_echo"
+- `--custom-function` - comma separated custom function name(s) to check like `pre_echo`
```
Comment (grogy) [inline README.md:51]: This commit can be squash with implementation

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,2 @@
 /vendor/
 /composer.lock
-.idea
```
Comment (grogy) [inline .gitignore]: This commit can be squashed with implementing commit

File: tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<?php
```
Comment (grogy) [inline tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php:1]: Please, why isn't this commit included in implementation? I mean that will be better for people who read Git history :)

File: tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+use JakubOnderka\PhpVarDumpCheck;
+
+class CustomFunctionTest extends PHPUnit_Framework_TestCase
+{
+    public function __construct()
```
Comment (grogy) [inline tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php]: Empty construction we can remove, what do you mean?

File: tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    {
+
+    }
+
+
```
Comment (grogy) [inline tests/JakubOnderka/PhpVarDumpCheck/CustomFunctionTest.php:26]: What about add test for none function founded? (I mean when code does not includes `functionName1`) It coverage all cases

File: src/Settings.php [hunk 7]
```diff
@@ -70,8 +70,9 @@ class Settings
      */
     public static function parseArguments(array $arguments)
     {
-        $arguments = new ArrayIterator(array_slice($arguments, 1));
-        $setting = new self;
+        $arguments       = new ArrayIterator(array_slice($arguments, 1));
```
Comment (grogy) [inline src/Settings.php]: Please do not horizontal align code, it is not used more in the codebase

**Review by grogy — COMMENTED** [review]
Perfect! Thank you for your PR. It looks great.

Here are some comments for improve code and repository history.
1. Can you help with squash commits? We can call together (like whereby.com) and I can show you how do it. 
2. Can you rebase it to current master?
3. We can add changelog line, it is perfect new feature.

**Comment by umutphp** [comment]
Hi @grogy ,

Many thanks for the review and feedbacks. This was a very old PR coming from the old repo :). I will fix the problems and make the PR ready for merge asap.

**Comment by umutphp** [comment]
I think I need to close this PR and come with a clean PR.
